src/main/diagnostics/limaDarwin.ts

```diff
@@ -0,0 +1,47 @@
+import semver from 'semver';
+
+import { spawnFile } from '@/utils/childProcess';
+import Logging from '@/utils/logging';
+
+import type { DiagnosticsCategory, DiagnosticsChecker } from './diagnostics';
+
+const console = Logging.diagnostics;
+
+/**
+ * CheckLimaDarwin version checks for compatibility with the host machine.
+ */
+const CheckLimaDarwin: DiagnosticsChecker = {
+  id:       'LIMA_DARWIN_VERSION',
```

Comment:
The id should be more specific to the memory limit, I think, as we may have more checks relating to the macOS version in the future.

Reply:
But in that case, we'd just say "we don't support <12.4", rather than all this logic dealing with memory, right? It doesn't seem sustainable to have a giant pile of things that all require you to do the same thing to resolve it…

Reply:
Yeah, I don't know. Maybe we can leave as is, and if we ever have another similar check, it just has to create it's own unique id. They are not user-visible, so they don't really matter that much.

We should probably have a file somewhere that lists all ids, either verbatim, or the pattern for generated ones.